cheroot/test/test_https.py

```diff
@@ -0,0 +1,146 @@
+"""HTTPS Tests."""
+# -*- coding: utf-8 -*-
+# vim: set fileencoding=utf-8 :
+
+import os
+import pytest
+import ssl
+
+from cheroot.ssl.builtin import BuiltinSSLAdapter
+from cheroot.test import helper
+import cheroot
+import ddt
```

Comment:
@dpeschman we now rely on pytest, which means no need for ddt anymore

Reply:
Since this PR was originally drafted on unittest-based tests, I'd be inclined to accept this change, rather than asking the contributor to chase our changes.

Reply:
I think I could try rewriting this on top of https://github.com/cherrypy/cheroot/tree/experiment/ssl-tests (ref #95). And we also need solid design change of `makefile` module (ref #6).
Anyway, it's currently unmergable because of conflicts and syntax is sometimes incompatible with Python 2. (see my `super()` comment)

P.S. We could probably drop some of hardcoded certificates in favor of `trustme`, which seems to be providing a very smooth experience for TLS testing with fake trusted CA.

Reply:
I've resolved the conflicts and Python 2 syntax issues. Happy to see iterations on the code to eliminate ddt and remove unittest constructs.